README.md

```diff
@@ -61,57 +49,70 @@ This is a basic example on how to use the library. Based on a query the language
 ```python
 # imports
 import torch
-from transformers import GPT2Tokenizer
-from trl.gpt2 import GPT2HeadWithValueModel, respond_to_batch
-from trl.ppo import PPOTrainer
+from transformers import AutoTokenizer
+from trl import PPOTrainer, PPOConfig, AutoModelForCausalLMWithValueHead
+from trl.core import respond_to_batch
 
 # get models
-gpt2_model = GPT2HeadWithValueModel.from_pretrained('gpt2')
-gpt2_model_ref = GPT2HeadWithValueModel.from_pretrained('gpt2')
-gpt2_tokenizer = GPT2Tokenizer.from_pretrained('gpt2')
+gpt2_model = AutoModelForCausalLMWithValueHead.from_pretrained('gpt2')
+gpt2_model_ref = AutoModelForCausalLMWithValueHead.from_pretrained('gpt2')
+gpt2_tokenizer = AutoTokenizer.from_pretrained('gpt2')
 
 # initialize trainer
-ppo_config = {'batch_size': 1, 'forward_batch_size': 1}
-ppo_trainer = PPOTrainer(gpt2_model, gpt2_model_ref, gpt2_tokenizer, **ppo_config)
+ppo_config = PPOConfig(
+    batch_size=1,
+    forward_batch_size=1
+)
 
 # encode a query
 query_txt = "This morning I went to the "
 query_tensor = gpt2_tokenizer.encode(query_txt, return_tensors="pt")
 
 # get model response
-response_tensor  = respond_to_batch(gpt2_model, query_tensor)
+response_tensor  = respond_to_batch(gpt2_model_ref, query_tensor)
 response_txt = gpt2_tokenizer.decode(response_tensor[0,:])
 
+# create a dummy dataset
+class DummyDataset(torch.utils.data.Dataset):
+    def __init__(self, query_data, response_data):
+        self.query_data = query_data
+        self.response_data = response_data
+
+    def __len__(self):
+        return len(self.query_data)
+
+    def __getitem__(self, idx):
+        return self.query_data[idx], self.response_data[idx]
+
+dummy_dataset = torch.utils.data.Dataset()
+
+min_length = min(len(query_tensor[0]), len(response_tensor[0]))
+
+dummy_dataset = DummyDataset(
+    [query_tensor[:, :min_length].squeeze(0) for _ in range(2)],
+    [response_tensor[:, :min_length].squeeze(0) for _ in range(2)],
+)
+
+# create a ppo trainer
+ppo_trainer = PPOTrainer(ppo_config, gpt2_model, gpt2_model_ref, gpt2_tokenizer, dummy_dataset)
+device = ppo_trainer.accelerator.device
+
 # define a reward for response
 # (this could be any reward such as human feedback or output from another model)
-reward = [torch.tensor(1.0)]
+reward = [torch.tensor(1.0).to(device)]
```

Comment:
We should let `accelerate` handle device placement internally. (Can address it in a follow up PR if you want)

Reply:
I think the reward would normally be from a preference model, so it would probably be on the correct device. But in this example that is not the case. I'll leave it to another PR.

Reply:
yes, discussed with @younesbelkada that we add some device placement steps to `PPOTrainer.step`.

Reply:
yes, this will be addressed in a PR right after merging this